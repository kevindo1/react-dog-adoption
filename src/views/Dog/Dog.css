body {
  font-family: cursive;
  font-weight: bold;
}
