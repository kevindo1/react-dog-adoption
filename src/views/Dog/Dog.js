import React from 'react';
import { useState, useEffect } from 'react';
import DogDetails from '../../components/DogDetails';
import { fetchDogsById, deleteDog } from '../../services/dogs';
import './Dog.css';
import { useParams } from 'react-router-dom/cjs/react-router-dom.min';

export default function Dog() {
  const [dog, setDog] = useState([]);
  const [loading, setLoading] = useState(true);
  const params = useParams();

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogsById(params.id);
      setDog(data);
      setLoading(false);
    };
    fetchData();
  }, [params.id]);

  if (loading) {
    return <h3>Loading</h3>;
  }
  const handleDelete = async (e) => {
    e.preventDefault();
    await deleteDog(dog.id);
    window.location.replace('/dogs');
  };

  return (
    <>
      <div>
        <DogDetails dog={dog} handleDelete={handleDelete} />
      </div>
    </>
  );
}
