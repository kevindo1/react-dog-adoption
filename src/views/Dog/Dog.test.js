import { render, screen } from '@testing-library/react';
import Dog from './Dog';
import { MemoryRouter, Route } from 'react-router-dom';

test('renders individual dog', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/12']}>
      <Route path="/dogs/:id" component={Dog} />
    </MemoryRouter>
  );
  await screen.findByText('Name: Thor');

  expect(container).toMatchSnapshot();
});
