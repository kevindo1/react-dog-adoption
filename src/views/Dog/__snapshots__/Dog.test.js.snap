// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders individual dog 1`] = `
<div>
  <div>
    <section>
      <div>
        <a
          href="/dogs/12"
        >
          <img
            src="https://placedog.net/500?id=3"
          />
        </a>
        <p>
          Name: 
          Thor
        </p>
        <p>
          Age: 
          4
        </p>
        <p>
          Breed: 
          Asguardian
        </p>
        <p>
          God of Thunder
        </p>
        <a
          href="/dogs/12/edit"
        >
          Edit
        </a>
        <button>
          Delete
        </button>
      </div>
    </section>
  </div>
</div>
`;
